### PR TITLE
Cluster creation improvements

### PR DESCRIPTION
**Pull request by lucas-giaco**
### Changelog
* Added @saaguero's changes to this PR
* Removed unused variables
* Encrypted secrets at rest by default
* Encrypted nodes root volume by default
* Enabling CSI driver StorageClasses by default
* Added `settings.json` file for vscode
* Added `CODEOWNERS` file
* Adding github actions
* Adding pre-commit hooks

### CODE REVIEW

File: cluster.tf
Comment (saaguero) [inline cluster.tf:37]: I've read that if you define the `ip_allocation_policy` as we do in this file this is taken as default `VPC_NATIVE`. Nevertheless, I think it won't hurt if we keep it explicitly like this.

File: cluster.tf
Comment (saaguero) [inline cluster.tf]: Should we revert this before merging back to master?

File: cluster.tf
Comment (saaguero) [inline cluster.tf:83]: I would say that since you are introducing this default here, make a variable out of it with the default value as "false".

File: cluster.tf
Comment (saaguero) [inline cluster.tf:95]: Seems like this is disabled by default: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#gce_persistent_disk_csi_driver_config

Do you know what are the implications of having it disabled? Probably it still works but using an old API or something.

Asking because to check if we want to include this in other projects.

File: network.tf
Comment (saaguero) [inline network.tf]: Probably you can remove this part of the "connector" as it was specific to other projects.

File: cluster.tf
Comment (lucas-giaco) [reply cluster.tf]: Yes, great catch thanks!

File: network.tf
Comment (lucas-giaco) [reply network.tf]: I Will do, thanks!

**Comment by jahernandezmartinez13** [comment]
There's a typo in the versions.tf name
Currently appears as `verstions.tf`

File: cluster.tf
Comment (jahernandezmartinez13) [inline cluster.tf:64]: Nice addition

File: cluster.tf
Comment (jahernandezmartinez13) [reply cluster.tf:37]: Agree with Santi, we can keep it for the sake of clarity.
We just need to remove the TODO comment

File: cluster.tf
Comment (jahernandezmartinez13) [reply cluster.tf:95]: Reading about it, it seems the future way of connecting volumes.
https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/gce-pd-csi-driver
I'd go with it enabled

File: data.tf
Comment (jahernandezmartinez13) [inline data.tf]: What do you mean?
I see the local variable master_version being used

File: variables.tf
Comment (jahernandezmartinez13) [inline variables.tf]: I'd set as default subnet range the initial Class A address (10.0.0.0)

File: nodes.tf
Comment (jahernandezmartinez13) [inline nodes.tf]: We should include the disk type as variable and set as default a SSD with very low size 10GB as our MS are stateless, nothing but the code is stored

Comment (jahernandezmartinez13) [reply nodes.tf]: Reading the K8s minimum requirements, maybe 10GB is too low
```Minimum 15 GB hard disk space for the file system containing```
We can set 20GB and compare the costs against 100GB non-ssd https://cloud.google.com/products/calculator to make the final decision

File: data.tf
Comment (lucas-giaco) [reply data.tf]: Sorry, that was a comment added while making changes. I'll remove it

File: nodes.tf
Comment (lucas-giaco) [reply nodes.tf]: As per the calculator, a 100Gb regional pd-standard disk cost U$D 8 a month, and a 20Gb regional pd-ssd disk costs U$D 6.80.

IMHO, I'll go with the 100Gb pd-standard disk, because we also need to take into account these variables:
* The number and size of the docker images stored in the node
* The size and rotation of the pod logs (despite if we're logging to stdout those logs are stored in the node)

Maybe we could shrink it to 80Gb that costs U$S 6,40 a month

Comment (saaguero) [reply nodes.tf]: I like the default of 100 GB (which is I believe the value used by the GCP module.

Comment (jahernandezmartinez13) [reply nodes.tf]: My intention was to improve the performance changing to SSDs, changing the size was just to reduce the cost of the SSD only
What would be the minimum disk to set for SSD? 40GB?
How it will compare in cost terms against the current solution (HDD-100GB)?

File: variables.tf
Comment (jahernandezmartinez13) [inline variables.tf:31]: Is this variable a zone or an array of zones?
The description says the only zone

Also I think we're missing to setup un the nodes the node_locations reading from this variable

File: data.tf
Comment (jahernandezmartinez13) [inline data.tf:26]: I'd prefer to keep all the data and local variables in one file.

File: .github/workflows/lint.yaml
Comment (saaguero) [inline .github/workflows/lint.yaml:50]: Cheers for including super-linter! :-D

File: variables.tf
Comment (lucas-giaco) [reply variables.tf:31]: Done, I used the same approach to determine the zones to use than the official module

**Review by saaguero — APPROVED** [review]
Great changes!

**Review by jahernandezmartinez13 — APPROVED** [review]

**Pull request merged**
